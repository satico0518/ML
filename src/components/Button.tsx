import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ text }: { text: string }) => <div className='custom-button'>{text}</div>;

Button.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Button;
