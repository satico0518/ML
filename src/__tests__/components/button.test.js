import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../components/Button';

describe('button component', () => {
  let buttonText;
  beforeEach(() => {
    buttonText = 'Vender';
  });
  test('should match snapshot', () => {
    const wrapper = shallow(<Button text={buttonText} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.custom-button').text()).toBe(buttonText)
  });
});
