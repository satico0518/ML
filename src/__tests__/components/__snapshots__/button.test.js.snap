// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component should match snapshot 1`] = `
<div
  className="custom-button"
>
  Vender
</div>
`;
